Composer/packages/types/src/diagnostic.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export interface IRange {
+  start: IPosition;
+  end: IPosition;
+}
+
+export interface IPosition {
+  line: number;
+  character: number;
+}
+
+export enum DiagnosticSeverity {
+  Error = 0,
+  Warning = 1,
+  Information = 2,
+  Hint = 3,
+}
+
+export interface IDiagnostic {
```

Comment:
convert these interfaces to types?

Reply:
This needs to be an interface because a class is implemented in `@bfc/shared`

Reply:
I looked into this after I saw those classes and it turns out you can actually implement types with classes:

```ts
type MyType = {
  someProp: boolean;
};

class MyClass implements MyType {
  someProp: boolean = true;
}
```

Reply:
Oh nice! I'll update then.